src/backend/partaj/core/api/referral.py

```diff
@@ -96,41 +98,96 @@ def get_permissions(self):
 
     def create(self, request, *args, **kwargs):
         """
-        Create a new referral as the client issues a POST on the referrals endpoint.
+        Create a empty referral as the client issues a POST on the referrals endpoint.
         """
-        form = ReferralForm(
-            {
-                # Add the currently logged in user to the Referral object we're building
-                "users": [request.user.id],
-                **{key: value for key, value in request.POST.items()},
-            },
-            request.FILES,
-        )
+        referral = models.Referral.objects.create()
+        referral.users.set([request.user])
+        referral.save()
+        return Response(data=ReferralSerializer(referral).data)
 
-        if form.is_valid():
+    def update(self, request, *args, **kwargs):
+        """
+        Update an existing referral.
+        """
+        referral = self.get_object()
+
+        referral.context = request.data.get("context")
+        referral.context = request.data.get("question")
+        referral.object = request.data.get("object")
+        referral.prior_work = request.data.get("prior_work")
+        referral.urgency_explanation = request.data.get("urgency_explanation")
+
+        if request.data.get("topic"):
+            try:
+                topic = models.Topic.objects.get(id=request.POST["topic"])
+                referral.topic = topic
+            except models.Topic.DoesNotExist:
+                return Response(
+                    status=400,
+                    data={
+                        "urgency_level": [
+                            f"{request.data.get('Topic')} is not a valid topic."
+                        ]
+                    },
+                )
+
+        referral.users.clear()
```

Comment:
IMO this is not necessary. We will probably prefer to reuse the multiple users widget from the referral detail view so we can forgo this on the referral form API route itself.